From cb7df46774e97de750e83aeca2d9035f919396f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20Stelmach?= <l.stelmach@samsung.com>
Date: Thu, 8 May 2014 12:57:26 +0200
Subject: [PATCH] core: check the right variable for failed open()

(cherry picked from commit cd7affaeea16d3904354b810a292e594dfef25dd)
(cherry picked from commit d7c1de4f3635830d28d6c61f3d1cae2a9474ffd8)
---
 src/core/cgroup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/cgroup.c b/src/core/cgroup.c
index 32e2599a2c..c215a86aad 100644
--- a/src/core/cgroup.c
+++ b/src/core/cgroup.c
@@ -661,7 +661,7 @@ int manager_setup_cgroup(Manager *m) {
                 close_nointr_nofail(m->pin_cgroupfs_fd);
 
         m->pin_cgroupfs_fd = open(path, O_RDONLY|O_CLOEXEC|O_DIRECTORY|O_NOCTTY|O_NONBLOCK);
-        if (r < 0) {
+        if (m->pin_cgroupfs_fd < 0) {
                 log_error("Failed to open pin file: %m");
                 return -errno;
         }
